tests/impl.cpp

```diff
@@ -3669,12 +3669,22 @@ result_t test_mm_cvtsd_si32(const SSE2NEONTestImpl &impl, uint32_t i)
 
 result_t test_mm_cvtsd_si64(const SSE2NEONTestImpl &impl, uint32_t i)
 {
-    return TEST_UNIMPL;
+    const double *_a = (const double *) impl.mTestFloatPointer1;
+#if defined(__x86_64__)
+    int64_t _c = (int64_t) round(_a[0]);
```

Comment:
The use of `round` is not generic here. You should explain.

Reply:
I think we set the different rounding mode in NEON and SSE.

### _mm_cvtsd_si64:
[`Converts the lower DP FP value of a to a 64-bit signed integer value according to the current rounding mode.`](
https://software.intel.com/content/www/us/en/develop/documentation/cpp-compiler-developer-guide-and-reference/top/compiler-reference/intrinsics/intrinsics-for-intel-streaming-simd-extensions-2-intel-sse2/integer-intrinsics/conversion-intrinsics-1.html)

### vcvt_s64_f64
`vcvt_s64_f64` is using rounding to zero.
https://developer.arm.com/architectures/instruction-sets/simd-isas/neon/intrinsics?search=vcvt_s64_f64

Reply:
Add comments in the test suite.

Reply:
IEEE 754 has 5 rounding rules, which are

* Round to nearest, ties to even
* Round to nearest, ties away from zero
* Round toward 0
* Round toward +∞
* Round toward -∞

Reference: https://en.wikipedia.org/wiki/IEEE_754#Rounding_rules

In the previous development, I found out the SSE rounding mode would be `Round to nearest, ties to even`.
However, I guess this might be the default rounding mode based on the description
```
_mm_cvtsd_si64:
Converts the lower DP FP value of a to a 64-bit signed integer value according to the current rounding mode.
```

NEON offers different rounding mode intrinsics, such as `vcvt`, `vcvtn` , `vcvtm` , `vcvtp` and `vcvta`.
Can we use the proper intrinsic to make the test case simpler?

Reply:
I wonder even the [C implementation](https://github.com/DLTcollab/sse2neon/pull/327/files#diff-4e3d6805bace5270202c37d5363eded99ce96bea91040c8123cdc16c4f858d36R5742) on ARM is using rounding to zero

Reply:
@JishinMaster, can you comment this?

Reply:
I got some idea here 
https://github.com/DLTcollab/sse2neon/blob/master/sse2neon.h#L5495

Reply:
I believe that the [C implementation](https://github.com/DLTcollab/sse2neon/pull/327/files#diff-4e3d6805bace5270202c37d5363eded99ce96bea91040c8123cdc16c4f858d36R5742) is wrong.
Anyway, the `_mm_cvtsd_si64()` would be affect by `_MM_SET_ROUNDING_MODE()`.
The test should include it to test 4 different rounding modes.